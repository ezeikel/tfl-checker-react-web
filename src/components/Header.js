import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import Nav from "./Nav";

const Wrapper = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  margin-bottom: 56px;
  padding: var(--spacing-large);
`;

const Logo = styled.section`
  font-size: 32px;
  font-weight: bold;
  font-family: var(--primary-font-family);
  a {
    color: var(--color-white);
  }
`;

const Header = () => {
  return (
    <Wrapper>
      <Logo>
        <Link to="/">TfL Checker</Link>
      </Logo>
      <Nav />
    </Wrapper>
  );
};

export default Header;
