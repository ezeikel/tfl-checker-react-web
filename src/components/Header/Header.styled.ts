import styled from "styled-components";

export const Wrapper = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  margin-bottom: 56px;
  padding: var(--spacing-large);
`;

export const Logo = styled.section`
  font-size: 32px;
  font-weight: bold;
  font-family: var(--primary-font-family);
  a {
    color: var(--color-white);
  }
`;
