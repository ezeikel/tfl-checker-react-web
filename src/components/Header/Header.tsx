import { Link } from "react-router-dom";
import Nav from "../Nav/Nav";
import { Wrapper, Logo } from "./Header.styled";

const Header = () => {
  return (
    <Wrapper>
      <Logo>
        <Link to="/">TfL Checker</Link>
      </Logo>
      <Nav />
    </Wrapper>
  );
};

export default Header;
