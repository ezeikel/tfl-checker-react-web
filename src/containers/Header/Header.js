import React from 'react';

import Nav from './Nav/Nav';

const Header = () => {
    return (
        <header className="header row">
            <Nav />
        </header>
    )
}

export default Header;
