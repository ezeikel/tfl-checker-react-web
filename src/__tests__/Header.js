import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

test('renders without crashing', () => {
  const { container } = render(
    <Router>
      <Header />
    </Router>
  );
  expect(container).toMatchSnapshot();
});