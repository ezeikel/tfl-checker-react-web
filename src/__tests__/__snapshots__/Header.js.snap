// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <header
    class="sc-AxiKw haLAnR"
  >
    <section
      class="sc-AxhCb fZhQsp"
    >
      <a
        href="/"
      >
        TfL Checker
      </a>
    </section>
    <nav
      class="sc-AxjAm iMxqKn"
    >
      <ul
        class="sc-AxirZ cBkftF"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Status updates
          </a>
        </li>
        <li>
          <a
            href="/trip-planner"
          >
            Plan a journey
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
